### ISSUE TITLE
Hume's rainbow cave: Octachoral test is broken

### ISSUE THREAD

**Issue by Wuzzy2** [SuperTux/addons#10]
Bug report for Hume's rainbow cave in SuperTux 0.4.0:

The level “Octachoral test” seems to be broken. If I enter a wrong answer the developer console opens and reports a warning (sector.cpp:499, “Squirrel error: Couldn't start script (attempt to call 'integer')”). Also, Tux is then forever trapped in the room when this happens. Sometimes the same error message appears on going through one of the pathways, but I could not reconstruct that.

More general feedback here: http://forum.freegamedev.net/viewtopic.php?f=69&t=2450&p=67138#p67075

**Comment by maxteufel** [comment]
Assigning all of #8, #9 and #10 to @Hume2. If they don't want to fix that, sorry. We don't provide official support for add-ons (else they would be part of the game).

**Comment by Wuzzy2** [comment]
Well, LMH told me to post the bug reports here to “increase visibility”, so I did so.

**Comment by Hume2** [comment]
I'm not fixing this, it will be intended behaviour.

**Comment by Hume2** [comment]
I'll edit the infoblock message, but this feature will be still there.

**Comment by Wuzzy2** [comment]
Wait, what? An error message is never a feature. Why is this a WONTFIX?

**Comment by LMH0013** [comment]
I agree.  There is an error message, then there is still a bug in the script.

In addition, trapping a player on a level without a way to finish/die is just bad form, this should be fixed.

**Comment by Hume2** [comment]
Firstly, the player is not trapped, there is still one way to continue. Secoundly, I've changed the message in the infoblock at the begining, so now it's obvious.
That error message must be caused by SuperTux, because there's no word "integer" in whole script.

So I'm not fixing that.

**Comment by LMH0013** [comment]
So you're right, there is still a way.  Although I'd hardly say anything is obvious from that message (nor can you complete the level if you answer everything wrong, you just end up alternate between two rooms, but I guess that's fair).  

Another thing I noticed is that if you manage to return to a room that you've already been the "wrong"/"correct" message appears instead of the question, making it exceedingly difficult to get the answer correct.  Sorry to point out another bug, but those pesky players are always finding new ways to break things.

A script error has to be related to the script.  What likely happened is that some of the syntax/structure of squirrel changed when it was updated (if the error wasn't originally there anyway).  Looks like you're trying to call an integer as a function, so no, "integer" would not need to be written in the script to cause the error.

**Comment by Hume2** [comment]
I'll look into the script again. Anyway typing zero at the beginning shouldn't affect the answer AFAIK.
